apps/web/src/pages/auth/LoginPage.tsx

```diff
@@ -6,13 +6,21 @@ import { AuthContext } from '../../store/authContext';
 import { LoginForm } from '../../components/auth/LoginForm';
 import AuthLayout from '../../components/layout/components/AuthLayout';
 import AuthContainer from '../../components/layout/components/AuthContainer';
+import useVercelIntegration from '../../api/hooks/use-vercel-integration';
+import { useMantineTheme, Loader, Paper } from '@mantine/core';
+import { colors, Text } from '../../design-system';
 
 export default function LoginPage() {
+  const theme = useMantineTheme();
   const { setToken, token } = useContext(AuthContext);
   const navigate = useNavigate();
   const [params] = useSearchParams();
   const queryToken = params.get('token');
   const source = params.get('source');
+  const code = params.get('code');
+  const next = params.get('next');
```

Comment:
This check should be part of `apps/web/src/api/hooks/use-vercel-integration.ts` instead because it is used multiple times across the codebase (DRY-Principle)

Reply:
I think you're right,performing the check and getting the final value from the useVercelIntegration hook would be better,
Thanks :)